static/app/components/events/highlights/editHighlightsModal.tsx

```diff
@@ -331,9 +338,14 @@ export default function EditHighlightsModal({
   project,
   closeModal,
 }: EditHighlightsModalProps) {
-  const [highlightContext, setHighlightContext] =
-    useState<HighlightContext>(prevHighlightContext);
-  const [highlightTags, setHighlightTags] = useState<HighlightTags>(prevHighlightTags);
+  const {replay, ...rest} = prevHighlightContext;
+  const [highlightContext, setHighlightContext] = useState<HighlightContext>({
+    ...(replay && {replay: replay?.filter(k => k !== 'replay_id')}),
+    ...rest,
+  });
+  const [highlightTags, setHighlightTags] = useState<HighlightTags>(
```

Comment:
this removes `replayId` from the preview UI up top when in the highlight modal, if the user has previously pinned it